File: list_models.js

import { GoogleGenerativeAI } from "@google/generative-ai";
import dotenv from 'dotenv';
import fs from 'fs';
dotenv.config();

async function listModels() {
    const apiKey = process.env.GOOGLE_API_KEY || process.env.GEMINI_API_KEY;
    if (!apiKey) {
        console.error("No API KEY found in env variables.");
        return;
    }

    // const genAI = new GoogleGenerativeAI(apiKey);

    try {
        console.log("Attempting to list models via API key starting with: " + apiKey.substring(0, 5) + "...");

        // Using fetch directly for ListModels endpoint
        const url = `https://generativelanguage.googleapis.com/v1beta/models?key=${apiKey}`;
        const response = await fetch(url);
        const data = await response.json();

        let output = "=== AVAILABLE MODELS ===\n";
        if (data.models) {
            data.models.forEach(m => {
                if (m.name.includes("gemini")) {
                    output += `- ${m.name} [${m.supportedGenerationMethods.join(', ')}]\n`;
                }
            });
        } else {
            output += "Error listing models: " + JSON.stringify(data);
        }

        fs.writeFileSync('models.txt', output);
        console.log("Models written to models.txt");

    } catch (error) {
        console.error("Error:", error);
    }
}

listModels();

File: list_models.py
import google.generativeai as genai
import os
from dotenv import load_dotenv

load_dotenv()
api_key = os.getenv('GEMINI_API_KEY')
genai.configure(api_key=api_key)


print(f"Checking available models for key ending in ...{api_key[-4:]}...\n")

with open('models_list_utf8.txt', 'w', encoding='utf-8') as f:
    try:
        for m in genai.list_models():
            line = f"Model: {m.name}\n   - Methods: {m.supported_generation_methods}\n"
            f.write(line)
            if 'image' in m.name.lower() or 'vision' in m.name.lower():
                 f.write(f"   - [POTENTIAL IMAGE MODEL]\n")
            f.write("-" * 20 + "\n")
            print(line.strip()) # Also print to console
    except Exception as e:
        error_msg = f"Error listing models: {e}"
        print(error_msg)
        f.write(error_msg)
